### PR TITLE
V3 vault deployment script

### PR DESCRIPTION
**Pull request by jubeira**
# Deployment preparation: `V3 Vault`

Build-info: https://github.com/balancer/balancer-v3-monorepo/commit/25d73b3d091f5dde943ad6b7d90db9569222510d.

Please check artifact by artifact; they come from the same package but may have different compiler settings.

## Deployment preparation tasks checklist:

- [x] Deployments `README` is updated with the new task name and link
- [x] Task `README` has a description and links to the artifacts
- [x] `build-info` is updated with artifacts compiled at the specified commit
- [x] Artifacts are generated from `build-info`
- [x] New task has a working fork test 

## Code checklist:

- [x] The diff is legible and has no extraneous changes 
- [x] Complex code has been commented in tests and scripts
- [x] Fork test block number and test cases are appropriate for this deployment
- [x] The base branch is either `master`, or there's a description of how to merge

## Issue Resolution

N/A

### CODE REVIEW

File: v3/tasks/v3-vault-factory/test/task.fork.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    const vaultAsExtension = vaultExtension.attach(vault.address);
+    expect(await vaultAsExtension.getProtocolFeeController()).to.be.equal(protocolFeeController.address);
+  });
+
+  it('checks vaultAdmin constants', async () => {
+    expect(await vaultAdmin.getMinimumTradeAmount()).to.be.equal(1e6);
+    expect(await vaultAdmin.getMinimumWrapAmount()).to.be.equal(1e4);
+    expect(await vaultAdmin.getMinimumPoolTokens()).to.be.equal(2);
+    expect(await vaultAdmin.getMaximumPoolTokens()).to.be.equal(8);
+    expect(await vaultAdmin.getMaximumPoolTokens()).to.be.equal(8);
```
Comment (EndymionJkb) [inline v3/tasks/v3-vault-factory/test/task.fork.ts:50]:
```suggestion
```
Duplicated line

File: v3/tasks/v3-vault-factory/input.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+const vaultAdminArtifact = Vault.artifact('VaultAdmin');
+const vaultAdminCreationCode = vaultAdminArtifact.bytecode;
+
+const salt = '0x3877188e9e5da25b11fdb7f5e8d4fdddce2d22707ba04878a8e14700dd46fa82';
+const targetVaultAddress = '0xbA1333333333a1BA1108E8412f11850A5C319bA9';
+
+export default {
+  Authorizer,
+  pauseWindowDuration: 4 * MONTH * 12,
+  bufferPeriodDuration: 3 * MONTH,
```
Comment (joaobrunoah) [inline v3/tasks/v3-vault-factory/input.ts:35]:
```suggestion
  bufferPeriodDuration: 6 * MONTH,
```

This should be 6, right?

**Review by joaobrunoah — APPROVED** [review]
Just fixing the buffer period, but LGTM!

**Review by joaobrunoah — DISMISSED** [review]
I verify that the artifacts match the commit 25d73b3d091f5dde943ad6b7d90db9569222510d of Balancer's v3 monorepo.

File: v3/tasks/20241204-v3-vault/index.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    ethers.utils.keccak256(input.vaultAdminCreationCode),
+  ];
+
+  const vaultFactory = await task.deployAndVerify('VaultFactory', vaultFactoryArgs, from, force);
+
+  const vaultAddress = await vaultFactory.getDeploymentAddress(input.salt);
+
+  // if (vaultAddress !== input.targetVaultAddress) {
+  //   throw Error('Incorrect target address');
+  // }
```
Comment (EndymionJkb) [inline v3/tasks/20241204-v3-vault/index.ts]: Debug code
```suggestion
```

**Review by EndymionJkb — APPROVED** [review]
I confirm that the main repo commit is correct, and that the artifacts match what's in this repo.

**Review by joaobrunoah — APPROVED** [review]

**Pull request merged**